edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3185,7 +3185,10 @@ private Double calculateCategoryScore(String studentUuid, Long categoryId, List<
 			Assignment assignment = gradeRecord.getAssignment();
 						
 			// remove if not for this category (rule 1)
-			if(assignment.getCategory() != null && categoryId.longValue() != assignment.getCategory().getId().longValue()){
+			if(assignment.getCategory() == null){
+				return true;
+			}
+			if(categoryId.longValue() != assignment.getCategory().getId().longValue()){
```

Comment:
Is this comparison of long value equality correct?

http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Long.html#equals%28java.lang.Object%29